tfbuilder/cases.go

```diff
@@ -119,3 +120,31 @@ func CreatePolicyAndModifyFieldsOnIt(
 		},
 	}
 }
+
+func NotImportedResourceShouldErrorOutWithMeaningfulMessage(
+	providerFactory map[string]func() (tfprotov6.ProviderServer, error),
+	builder ModifyPolicyBuilder,
+	mtp *PolicyBuilder,
+	preConfigFn func(),
+) resource.TestCase {
+	expectedErr, _ := regexp.Compile(`MeshTrafficPermission already exists`)
+
+	return resource.TestCase{
+		ProtoV6ProviderFactories: providerFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: builder.Build(),
+			},
+			{
+				PreConfig:   preConfigFn,
+				Config:      builder.AddPolicy(mtp.WithSpec(AllowAllTrafficPermissionSpec)).Build(),
+				ExpectError: expectedErr,
```

Comment:
this is the part where we tell the testing framework to check for that error message, the full message is:

```
MeshTrafficPermission already exists

A resource with the name xxx already exists in the mesh yyy - to be managed via Terraform it needs to be imported first"
```

but I think the framework only allows matching on summary

see:

https://github.com/Kong/shared-speakeasy/blob/main/generators/resource_plan_modifier/template.go.tmpl#L87-L89